DogScepterLib/Core/GMChunk.cs

```diff
@@ -1,155 +1,151 @@
+        { "TGIN", typeof(GMChunkTGIN) },
+        { "FONT", typeof(GMChunkFONT) },
+        { "SPRT", typeof(GMChunkSPRT) },
+        { "ACRV", typeof(GMChunkACRV) },
+        { "FUNC", typeof(GMChunkFUNC) },
+        { "VARI", typeof(GMChunkVARI) },
+        { "SCPT", typeof(GMChunkSCPT) },
+        { "TAGS", typeof(GMChunkTAGS) },
+        { "ROOM", typeof(GMChunkROOM) },
+        { "OBJT", typeof(GMChunkOBJT) },
+        { "TMLN", typeof(GMChunkTMLN) },
+        { "GLOB", typeof(GMChunkGLOB) },
+        { "GMEN", typeof(GMChunkGMEN) },
+        { "SHDR", typeof(GMChunkSHDR) },
+        { "CODE", typeof(GMChunkCODE) },
+        { "SEQN", typeof(GMChunkSEQN) },
+        { "FEDS", typeof(GMChunkFEDS) },
+    };
+    public static readonly Dictionary<Type, string> ChunkMapReverse = ChunkMap.ToDictionary(x => x.Value, x => x.Key);
+
+    public override void Serialize(GMDataWriter writer)
+    {
+        base.Serialize(writer);
 
-            // Write all the sub-chunks
-            for (int i = 0; i < ChunkNames.Count; i++)
-            {
-                GMChunk chunk;
-                if (Chunks.TryGetValue(ChunkNames[i], out chunk))
-                {
-                    writer.Data.Logger?.Invoke($"Writing {ChunkNames[i]} at {writer.Offset:X}");
+        int beg = writer.BeginLength();
 
-                    // Write chunk name, length, and content
-                    writer.Write(ChunkNames[i].ToCharArray());
-                    int chunkBeg = writer.BeginLength();
-                    chunk.Serialize(writer);
+        // Write all the sub-chunks
+        for (int i = 0; i < ChunkNames.Count; i++)
+        {
+            GMChunk chunk;
+            if (!Chunks.TryGetValue(ChunkNames[i], out chunk)) continue;
+
+            writer.Data.Logger?.Invoke($"Writing {ChunkNames[i]} at {writer.Offset:X}");
 
-                    // If not the last chunk, apply 16-byte padding if necessary
-                    if (writer.Data.VersionInfo.AlignChunksTo16 && i != (ChunkNames.Count - 1))
-                        writer.Pad(16);
+            // Write chunk name, length, and content
+            writer.Write(ChunkNames[i].ToCharArray());
+            int chunkBeg = writer.BeginLength();
+            chunk.Serialize(writer);
 
-                    writer.EndLength(chunkBeg);
-                }
-            }
+            // If not the last chunk, apply 16-byte padding if necessary
+            if (writer.Data.VersionInfo.AlignChunksTo16 && i != (ChunkNames.Count - 1))
+                writer.Pad(16);
 
-            writer.EndLength(beg);
+            writer.EndLength(chunkBeg);
         }
 
-        public override void Deserialize(GMDataReader reader)
+        writer.EndLength(beg);
+    }
+
+    public override void Deserialize(GMDataReader reader)
+    {
+        base.Deserialize(reader);
+
+        // Gather the names and offsets of the sub-chunks
+        ChunkNames = new List<string>();
+        //TODO: updated, but never used
```

Comment:
Uh... maybe this is unused because you deleted the code that used it?

Reply:
Oh whoops. Error from resolving a local merge conflict.